src/stdlib/File.ts

```diff
@@ -0,0 +1,87 @@
+import { Callable, ValueKind, BrsString, BrsBoolean } from "../brsTypes";
+import { Interpreter } from "../interpreter";
+import { Volume } from "memfs/lib/volume";
+
+/* Returns the brs-path uri scheme or an empty string if not found */
+export function parseVolumeName(path: string): string {
+    const volumeRegex = new RegExp("^([a-z]+):");
+    const volumePrefix = path.match(volumeRegex);
+    if (volumePrefix !== null) {
+        return volumePrefix[1];
+    } else {
+        return "";
+    }
+}
+
+/*
+ * Returns a memfs volume based on the brs path uri.  For example, passing in
+ * "tmp:///test.txt" will return the memfs temporary volume on the interpreter. 
+ * 
+ * Returns invalid in no appopriate volume is found for the path
+ */
+export function getVolumeByPath(interpreter: Interpreter, path: string): Volume | null {
+    const volumePrefix = parseVolumeName(path);
+    if (volumePrefix === "tmp") {
+        return interpreter.temporaryVolume;
+    } else {
+        return null;
+    }
+}
+
+/*
+ * Removed the scheme from the uri.
+ *    ex. "tmp:///test/test1.txt" -> "///test/test1.txt"
+ * 
+ * NOTE: memfs as well as other file systems correctly handle extraneous slashes
+ */
+function stripScheme(brsUri: string): string {
+    return brsUri.replace(parseVolumeName(brsUri) + ":", "");
+}
+
+/** Reads ascii file from file system. */
+export const ReadAsciiFile = new Callable(
+    "ReadAsciiFile",
+    {
+        signature: {
+            args: [
+                {name: "filepath", type: ValueKind.String}
+            ],
+            returns: ValueKind.String
+        },
+        impl: (interpreter: Interpreter, filepath: BrsString, text: BrsString) => {
+            const volume = getVolumeByPath(interpreter, filepath.value);
+            if (volume === null) {
```

Comment:
`tslint` is configured to allow `==` comparisons against `null`, because it matches both `null` and `undefined`.  Might be worth using that here.

Reply:
What's the advantage?

Reply:
Nothing major, but anything that accidentally returns `undefined` is also handled.